src/time/gadgets.cc

```diff
@@ -21,12 +22,12 @@ time_t squid_curtime = 0;
 time_t
 getCurrentTime()
 {
-#if GETTIMEOFDAY_NO_TZP
-    gettimeofday(&current_time);
-#else
+    using namespace std::chrono;
+    const auto now = system_clock::now();
 
-    gettimeofday(&current_time, nullptr);
-#endif
+    const auto ms = duration_cast<milliseconds>(now.time_since_epoch());
+    current_time.tv_sec = ms.count()/1000;
+    current_time.tv_usec = (ms.count()%1000)*1000;
```

Comment:
Official code uses system gettimeofday() precision that may go up to microseconds. PR code uses milliseconds instead. The difference can be up to three orders of magnitude, and many Squid transactions last less than a millisecond. This PR has to disclose and justify this change in precision but does neither.

IIRC, LFT_TIME_SUBSECOND (i.e. `%tu`) is a logformat code that can be used to log current time with varying precision:

```
%tu    -- 123 -- millisecond precision (default)
%.3tu  -- 123 -- millisecond precision (explicit)
%.6tu  -- 123456 -- microsecond precision
```

I have not studied all the necessary details, but I suspect that this PR code should be refactored to use microsecond precision (where available from system_clock::now()) for setting tv_usec data member, addressing this concern.

P.S. Other adjustments might be necessary, but I am focusing on this one big item for now.

<!-- sandbox: https://godbolt.org/z/oj11ofTe7 -->

Reply:
Thanks for catching that. On second thought we do not have to preserve the ms resolution at all. Should be fixed with the additional b8638046d9 refactor.